### PR TITLE
Eswe 721 add prison id and prison name

### PR DESCRIPTION
**Pull request by sacintha**

### CODE REVIEW

**Review by theDustRoom — APPROVED** [review]
LGTM

File: build.gradle.kts [hunk 1]
```diff
@@ -53,7 +53,7 @@ dependencies {
   implementation("org.jetbrains.kotlin:kotlin-reflect:1.6.20")
 
   // Database dependencies
-  runtimeOnly("org.flywaydb:flyway-core")
+  implementation("org.flywaydb:flyway-core")
```
Comment (petergphillips) [inline build.gradle.kts:56]: not sure why this is required - normally should just be `runtimeOnly` as you don't need to reference flyway in your code?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/data/jsonprofile/CIAGProfileDTO.kt [hunk 2]
```diff
@@ -16,6 +16,11 @@ data class CIAGProfileDTO(
   @Schema(description = "This is the ID of the inmate ", name = "offenderId", pattern = "^[A-Z]\\d{4}[A-Z]{2}\$", required = true)
   val offenderId: String,
 
+  @Schema(description = "This is the prision ID of the inmate ", name = "prisonId", pattern = "^[A-Z]{3}\$", required = true)
+  var prisonId: String?,
+
+  @Schema(description = "This is the prision Name of the inmate ", name = "prisonName", pattern = "^[A-Z]{3}\$", required = true)
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/data/jsonprofile/CIAGProfileDTO.kt:22]: typo in `prision`, also pattern is incorrect for prison name

File: src/main/resources/db/migration/V2_1__ADD_PRISON_NAME_AND_ID.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+alter tablea ciag_profile add column prison_id varchar(255);
```
Comment (petergphillips) [inline src/main/resources/db/migration/V2_1__ADD_PRISON_NAME_AND_ID.sql:1]: what is `tablea`?  `prison_id` should be varchar(8) instead.  Is this actually migration being tested?

Comment (nathanrussell-moj-digital) [reply src/main/resources/db/migration/V2_1__ADD_PRISON_NAME_AND_ID.sql:1]: This relates to my slack message earlier @sacintha 👍 
Screenshot 2023-10-16 at 11 43 14

**Pull request merged**